### ISSUE TITLE
Bun could not find a package.json file.

### ISSUE THREAD

**Issue by henriquealbert** [oven-sh/bun#5483]
### What version of Bun is running?

1.0.1+31aec4ebe325982fc0ef27498984b0ad9969162b

### What platform is your computer?

Darwin 23.0.0 arm64 arm

## Issue while trying to install packages using `bun install` in Meteor build

Steps to reproduce:
```bash
$ meteor build --directory ./dist
$ cd dist/bundle/programs/server
$ bun install


error: MissingPackageJSON
Bun could not find a package.json file.
```

### What is the expected behavior?

It should simply install the npm dependencies.

### What do you see instead?

![Screenshot 2023-09-15 at 14 04 33](https://github.com/oven-sh/bun/assets/59376543/012b2949-05ca-4e0c-9e5c-afb2a5042b56)

### Additional information

I created a repository and added the steps to reproduce the issue: https://github.com/henriquealbert/bun-meteor-build

**Comment by elwqnn** [comment]
upupup

**Comment by L422Y** [comment]
This is occurring for me when using local package sources:

```json
 "devDependencies": {
    "my-package": "file:/Users/larry/Development/oss/my-package"
  }
```
  
I've confirmed there is a valid package.json in the `my-package` directory

Everything works as expected with `pnpm` and `npm`

**Comment by pgmatheus** [comment]
In my case, I've added 'sudo,' and everything worked after.

**Comment by rafsawicki** [comment]
I actually ran into a similar issue, but with a docker build, where a package.json file was copied with incorrect user permissions. In mine (and I guess @pgmatheus) instance, the bug is more about providing a proper error message - the file exists but is not accessible for bun.

**Comment by Kitenite** [comment]
I ran into a similar issue, linking by file in package. package.json exist in dependency package. npm install works fine.
```
  "dependencies": {
          "my-package": "file://Users/path/to/package"
```

**Comment by fpigeonjr** [comment]
ran into similar issue with Docker and it was a permission error here is how we solved it:

```
# Dockerfile

COPY --chown=app:app ./package.json ./bun.lockb /home/app/
````

**Comment by garviand** [comment]
Running into this with GitHub actions. I'm not using any `file://` dependencies.

**Comment by kylekirkby** [comment]
My issue was that I was using a Git submodule but hadn't set the submodule up in a fresh clone. We should handle this in a cleaner way. If possible, warn users that the package was missing?

**Comment by smartrusorg** [comment]
Command working after installing latest node js 😁

**Comment by yxlwfds** [comment]
up, ubuntu, snap install bun-js,package.json has created,npm i is ok

**Comment by yxlwfds** [comment]
![image](https://github.com/oven-sh/bun/assets/2525362/bf082243-16a0-48c2-8cc9-2b026ab755dd)
删除不掉,装不上,用不了,哈哈

**Comment by Jarred-Sumner** [comment]
if you can use `sudo perf trace` or `strace` and report the output, that would help. Seeing what the system call is returning would tell us where the error is

**Comment by PaulJPhilp** [comment]
I had this problem today.  It turns out that I had this entry in my "dependencies" : ".", ".".  When I removed that entry, bun started to work again.  I don't know how that line got there, perhaps bun put it there at some point?

![image](https://github.com/oven-sh/bun/assets/138239846/581e5dd5-7573-42f9-a02a-d47419c9ccab)

**Comment by patrickjquinn** [comment]
Piling on here, where I'm pointing a dep at "file://" in the package.json I get the "missing package.json" message. No issue with yarn, npm or pnpm.

**Comment by alanpog** [comment]
I fixed this by using [bun link](https://bun.sh/docs/cli/link) instead of the `file:/` syntax. That seems to be the bun way to handle local deps.

**Comment by L422Y** [comment]
yep! "bun link" works better than any other solution for me as well

**Comment by PaulJPhilp** [comment]
Awesome, thanks!

**Comment by CommanderGL** [comment]
I am trying to install a custom cli and am getting this error when running this command: `bun add -g .` with this `package.json`:

```json

  "name": "cslang",
  "module": "src/index.ts",
  "type": "module",
  "scripts": {
    "dev": "bun src/index.ts"
  },
  "devDependencies": {
    "@types/bun": "latest"
  },
  "dependencies": {
    "citty": "^0.1.6",
    "consola": "^3.2.3"
  },
  "bin": {
    "cslang": "./src/index.ts"
  }
}
```

**Comment by fccjr24** [comment]
bun link didn't work for me when linking to a separate package in a monorepo

**Comment by ricardolucena** [comment]
Same problem here.

**Comment by ricardolucena** [comment]
ricardo@ricardo:/var/www/html/typebot.io$ bun i
error: Bun could not find a package.json file to install from
note: Run "bun init" to initialize a project

**Comment by ojvribeiro** [comment]
I'm also getting this when trying running bun in a GitHub Action.

**Comment by avdv** [comment]
I also ran into this when using bun 1.0.14. Although I was using `--no-save` bun tried to open the `package.json` file in read/write mode -- on a read-only filesystem (in a sandbox):
```
openat(AT_FDCWD, "/tmp/bazel-working-directory/_main/package.json", O_RDWR|O_NOCTTY|O_CLOEXEC) = -1 EROFS (Read-only file system)
```
Using bun 1.1.8 instead works. Just my 2 cents.

**Comment by PauloVBettio** [comment]
In my case, the `snap` installation of bun was causing this problem. Ditching it and installing the latest version with `curl` fixed this issue.

**Comment by u007** [comment]
i got this issue with bcrypt
